### ISSUE TITLE
mdns-publisher-worker-X pods fail to start

### ISSUE THREAD

**Issue by yprokule** [openshift-metal3/dev-scripts#458]
After applying https://github.com/openshift-metal3/dev-scripts/pull/453 _mdns-publisher-worker-X_ pods fail to start:
```
mdns-publisher-worker-0   0/1     Init:CrashLoopBackOff   8          18m
mdns-publisher-worker-1   0/1     Init:CrashLoopBackOff   8          18m
mdns-publisher-worker-2   0/1     Init:CrashLoopBackOff   8          17m
```
Error:
```
oc logs mdns-publisher-worker-0 -n openshift-kni-infra -c render-config
+ source /etc/kubernetes/static-pod-resources/clusterrc
++ NAME=local
++ APIURL=https://api.ostest.test.metalkube.org:6443
++ APIHOST=api.ostest.test.metalkube.org
++ DOMAIN=ostest.test.metalkube.org
++ BASE_DOMAIN=test.metalkube.org
+ CLUSTER_NAME=local
++ dig +noall +answer api.ostest.test.metalkube.org
++ awk '{print $NF}'
+ API_VIP=192.168.123.5
++ grep -v 'scope host'
++ ip addr show
++ grep -Po 'inet \K[\d.]+/[\d.]+'
++ xargs
+ IFACE_CIDRS='172.22.0.27/24 192.168.123.127/24 10.131.0.1/23'
++ /usr/local/bin/get_vip_subnet_cidr 192.168.123.5 '172.22.0.27/24 192.168.123.127/24 10.131.0.1/23'
/bin/bash: line 7: /usr/local/bin/get_vip_subnet_cidr: Is a directory
+ SUBNET_CIDR=
```

**Comment by celebdor** [comment]
Also found out that the workers clusterrc generation is inaccurate stemming from the fact that their kubeconfig puts the cluster name as `local` instead of the real cluster name.

```
	  name: ostest		  name: local
	contexts:		contexts:
	- context:		- context:
	    cluster: ostest		    cluster: local
	    user: kubelet		    user: kubelet
	  name: kubelet		  name: kubelet
	current-context: kubelet		current-context: kubelet
	preferences: {}		preferences: {}
```
Left master right worker

**Comment by celebdor** [comment]
will dig deeper later

**Comment by nixpanic** [comment]
This might fix it: https://github.com/nixpanic/dev-scripts/commit/bea63ec2a07d9fd606d40cd07a165f4ec014fc0a

I'm testing a deployment with that now, and will create a PR if it works.

**Comment by nixpanic** [comment]
My change is not sufficient, but the `render-config` initContainer gets a little further:
```
IOError: [Errno 2] No such file or directory: '/etc/kubernetes/static-pod-resources/config.template'
```
The template is called `worker-config.template`, adapting that in the manifest and trying again.

**Comment by nixpanic** [comment]
@celebdor I think this can be closed, #469 seems to have been sufficient.

**Comment by hardys** [comment]
Closing as https://github.com/openshift-metal3/dev-scripts/pull/469 merged, thanks @nixpanic
